### PR TITLE
Add `AzureContainerInstanceJob.kill` for flow run cancellation

### LINKED ISSUE #59: Flow Run Cancellation for `AzureContainerInstanceJob` infrastructure block
Prefect supports flow cancellation as of the 2.7.0 release, as explained in the release announcement. 

The Azure Container Instances infrastructure block should support the new cancellation.

When merged, PR #58 will resolve this issue.

### PR DESCRIPTION
**Pull request by rpeden**
This PR adds a `kill` method to `AzureContainerInstanceJob` to support flow run cancellation added in Prefect 2.7.0. See the PR in the `prefect` repository for further detail: https://github.com/PrefectHQ/prefect/pull/7666

Small updates to exception handling, log messages and type annotations were added to other parts of `AzureContainerInstanceJob` to make it more resilient, particularly when making Azure log fetch API calls that fail sporadically when running an underpowered container under heavy load.

## Example 
The only visible effect of these updates is that the user can click the big red Cancel button on the Prefect >= 2.7.0 Flow Run page:

Screen Shot 2022-12-01 at 10 51 50 PM

After confirming they are sure they want to cancel the flow run, the user can check the Azure Portal to confirm that was once a happily running flow:

Screen Shot 2022-12-01 at 10 54 16 PM

has now vanished, with its container group and all containers deleted:

Screen Shot 2022-12-01 at 10 54 42 PM

Closes #59.

### Checklist

- [x] This pull request references any related issue by including "Closes #<ISSUE_NUMBER>"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-azure/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] Summarized PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-azure/blob/main/CHANGELOG.md)

### CODE REVIEW

File: prefect_azure/container_instance.py [hunk 1]
```diff
@@ -299,6 +308,55 @@ async def run(
             identifier=created_container_group.name, status_code=status_code
         )
 
+    async def kill(self, container_group_name: str, grace_seconds: int = 30):
```
Comment (ahuang11) [inline prefect_azure/container_instance.py:311]: Should grace_seconds also use CONTAINER_GROUP_DELETION_TIMEOUT_SECONDS?

File: prefect_azure/container_instance.py [hunk 2]
```diff
@@ -299,6 +308,55 @@ async def run(
...
+        """
+        Kill a flow running in an ACI container group.
+        Args:
```
Comment (ahuang11) [inline prefect_azure/container_instance.py:314]: Nit:
```suggestion
        Kill a flow running in an ACI container group.

        Args:
```

File: prefect_azure/container_instance.py [hunk 3]
```diff
@@ -299,6 +308,55 @@ async def run(
...
+            container_group = aci_client.container_groups.get(
+                resource_group_name=self.resource_group_name,
+                container_group_name=container_group_name,
+            )
+        except ResourceNotFoundError:
+            # the container group no longer exists, so there's nothing to cancel
+            raise InfrastructureNotFound(
+                f"Cannot stop ACI job: container group {container_group_name} "
+                "no longer exists."
+            )
```
Comment (ahuang11) [inline prefect_azure/container_instance.py:342]: Maybe this to link the tracebacks?

```suggestion
        except ResourceNotFoundError as exc:
            # the container group no longer exists, so there's nothing to cancel
            raise InfrastructureNotFound(
                f"Cannot stop ACI job: container group {container_group_name} "
                "no longer exists."
            ) from exc
```

**Review by ahuang11 — COMMENTED** [review]
Thanks for your help!

**Review by ahuang11 — DISMISSED** [review]
Thanks so much for keeping feature parity for Azure!

File: prefect_azure/container_instance.py [hunk 4]
```diff
@@ -543,7 +649,7 @@ def _get_logs(
         client: ContainerInstanceManagementClient,
         container_group: ContainerGroup,
         max_lines: int = 100,
-    ) -> str:
+    ) -> Union[str, None]:
```
Comment (ahuang11) [inline prefect_azure/container_instance.py:652]: If I'm not mistaken:
```suggestion
    ) -> str:
```

**Review by desertaxle — APPROVED** [review]
LGTM!

**Review by ahuang11 — APPROVED** [review]

**Pull request merged**